docs/electrical/embedded/styleguide.md

```diff
@@ -1,3 +1,136 @@
+2. For further clarity, each side of a logical operator should be surrounded with parentheses, unless it is a single level operation.
+```c title="Example"
+((current_depth == target_depth) && (current_depth < pool_depth))
+
+(soft_arm && hard_arm)
+```
+
+#### Common Abreviations
+1. Unless it is an abbreviation in common use and understood widely among programmers, refrain from using abreviations as they can obfuscate the meaning of code.
+2. Common variable names, such as ```i``` for a loop counter can be used under the condition that it is of limited scope and purpose, and that the context it is in is exceedingly clear.
+
+#### Casts
+1. Should a cast be required, the relevant lines should be commented to explain assumptions made.
+
+#### Keywords
+1. The use of ```goto``` should be exceedingly limited to only cases where it is justifiably required, and no other solution can be found (including rewriting other code to make the ```goto``` unecessary).
+2. Uses of the ```continue``` keyword should be limited, but are allowed where it simplifies the code structure and makes it more legible.
+3. *The ```static``` keyword needs some though put towards its use cases, I don't know if anything needs to be defined for this.*
+4. ```const``` should be applied in all locations where appropriate. In some cases, it may be more applicable than a preprocessor directive, while the preprocessor directive may be more applicable in others.
+5. ```volatile``` should be used in all cases where a variable can be modified by anything outside of the normal control flow of the code (i.e. an ISR).
+
+
+### Comment Rules
+1. Both single (//) and multi (/*) comment styles are considered acceptable, and should be used as applicable. When multi-line comments are used, all lines should begin with an asterisk (\*)
+2. Multi-line comments should maintain a level of indentation with the leading asterisks vertical to each other, as shown below.
+3. One leading space should be put between the comment text and the comment symbol.
+4. If placed at the end of a line of code, all comments should be vertically aligned.
+```c
+/* comment text
+ * comment text
+ * comment text
+ */
+
+ // comment text
+```
+
+
+5. All comments should be written in coherent, complete sentences.
+6. Assume a competentcy with the language from a reader, and avoid explaining the obvious. (*You aren't trying to hit a comment count for an assignment, you're trying to actually be helpful to yourself and others in the future*)
+7. Single line comments should be used where the function of a single line may be ambiguous.
+
+*TODO: Consider whether it's necessary to require conformance to automatic documentation generation standards*
+
+#### Comment Locations
+1. Functions should be preceeded by a short comment block explaining the relevant values (inputs, outputs) and an extremely brief description of the purpose of the function.
+```c title="Example"
```

Comment:
Markdown formatting issue. Blank line between this code block and the numbered list **OR** indent the code block below the "1" item in the list